File: 5.implementation/redsun/src/main/webapp/static/js/services/CalendarDefinitionsService.js


/**
 * Controller for ${entity.name}
 **/


app.factory('calendarDefinitionService', function($http, $rootScope) {
	return {
	
		// Insert.
		add: function(calendarDefinition) {
			var urlServer = $rootScope.makePostURL("/calendardefinition/insert");
			return $http.post(urlServer, calendarDefinition);
		}
		
		// Update.
		, update: function(calendarDefinitionId, calendarDefinition) {
			var urlServer = $rootScope.makePostURL("/calendardefinition/update/" + calendarDefinitionId);
			return $http.put(urlServer, calendarDefinition);
		}
		
		// Delete.
		, delete: function(calendarDefinitionId) {
			var urlServer = $rootScope.makePostURL("/calendardefinition/delete/" + calendarDefinitionId);
			return $http.delete(urlServer);
		}
		
		// Get by Id.
		, getById: function(calendarDefinitionId) {
			var url = $rootScope.makeGetURL("/calendardefinition/getbyid/" + calendarDefinitionId);
			return $http.get(url);
		}
		
		// List for page and filter.
		, getsForPageAndFilter: function(itemsPerPage, pageNo, calendarDefinition) {
			var url = $rootScope.makeGetURL("/calendardefinition/list/page/filter/" + itemsPerPage + "/" + pageNo);
			return $http.get(url, { params: calendarDefinition });
		}
		
	}// return.
});

File: 5.implementation/redsun-service/src/main/java/com/redsun/service/service/CalendarDefinitionsService.java
package com.redsun.service.service;

import com.redsun.service.entities.CalendarDefinitions;
import com.redsun.service.entities.Result;

/**
 * CalendarDefinitions Service interface
 */
public interface CalendarDefinitionsService {
	
    // Insert.
	Result insert(final CalendarDefinitions calendarDefinitions) ;

    // Update.
	Result update(final CalendarDefinitions calendarDefinitions) ;

    // Delete.
	Result delete(final Integer id) ;

	// Get By Id.
	Result getById(final Integer id) ;

    // Exists.
	Result exists(final Integer id) ;

    // Count.
	Result count() ;

	// List for page and filter.
	Result listCalendarDefinitionsForPageAndFilter(final int itemsPerPage, final int pageNo, final CalendarDefinitions calendarDefinitions);

}
